open_spiel/python/examples/lewis_signaling_dqn.py

```diff
@@ -0,0 +1,255 @@
+# Copyright 2019 DeepMind Technologies Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""DQN example on Lewis Signaling Game.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import copy
+from absl import app
+from absl import flags
+import numpy as np
+from six.moves import range
+import tensorflow.compat.v1 as tf
+
+from open_spiel.python import rl_environment
+from open_spiel.python import rl_tools
+from open_spiel.python.algorithms import dqn
+
+FLAGS = flags.FLAGS
+
+# Env parameters
+flags.DEFINE_integer("num_states", 3, "Number of states and actions")
+flags.DEFINE_integer("num_messages", 3, "Number of messages")
+flags.DEFINE_string("payoffs", "1, 0, 0, 0, 1, 0, 0, 0, 1",
+                    "Payoffs to use ('random' for random [0, 1) payoffs)")
+
+# Alg parameters
+flags.DEFINE_bool("centralized", False, "Set to use centralized Q-Learning")
+flags.DEFINE_integer("num_episodes", 50000, "Number of train episodes")
+flags.DEFINE_float("step_size", 0.1, "Step size for updates")
+flags.DEFINE_float("eps_init", 1.0, "Initial value of epsilon")
+flags.DEFINE_float("eps_final", 0.0, "Final value of epsilon")
+flags.DEFINE_integer("eps_decay_steps", 49000,
+                     "Number of episodes to decay epsilon")
+
+# Misc paramters
+flags.DEFINE_integer("num_runs", 10, "Number of repetitions")
+flags.DEFINE_integer("log_interval", 100,
+                     "Number of episodes between each logging")
+flags.DEFINE_bool("plot", False, "Set to plot the graphs")
+
+
+def main(_):
+    game = "lewis_signaling"
+    num_players = 2
```

Comment:
Use the value of 2 as a Flag

Reply:
Since the game always has only 2 players, I just used a variable and not a flag.